src/semeio/forward_models/overburden_timeshift/ots.py

```
@@ -279,7 +305,12 @@ def geertsma_ts_simple(self, vintage_pairs):
             vintage_pairs, self.subsidence.eval_geertsma, "TS_SIMPLE"
         )
 
-    def _geertsma_ts_custom(self, vintage_pairs, subsidence_func, method_name):
+    def _geertsma_ts_custom(
+        self,
+        vintage_pairs: Sequence[ConstrainedList],
+        subsidence_func: Callable[..., float],
+        method_name: str,
+    ) -> tuple[Literal[0], list[Any]] | list[xtgeo.RegularSurface]:
```

Comment: The typing here is a bit wonky. The reason for this is that if `len(vintage_pairs) < 1` one returns a tuple `0, []`, which is not really consistent with the `list[xtgeo.RegularSurface]`. This typing also propagates since other functions use this function as well. This should probably raise an error instead, so the type of the output is `list[xtgeo.RegularSurface]`